devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package bmp180 controls a Bosch BMP180 device over I²C.
+//
+// Datasheet
+//
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+package bmp180
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/devices"
+)
+
+// Oversampling affects how much time is taken to measure pressure.
+type Oversampling uint8
+
+// Possible oversampling values.
+const (
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+)
+
+// Dev is an handle to a bmp180.
+type Dev struct {
+	d  conn.Conn
+	c  calibration
+	os Oversampling
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("BMP180{%s}", d.d)
+}
+
+const (
+	ctrlCommandTemperature = 0x0E
+	ctrlCommandPressure    = 0x14
+)
+
+// bit offsets in regCtrlMeas
+const (
+	ctrlMeasurementControl = 0
+	ctrlStartConversion    = 5
+	ctrlOversampling       = 6
+)
+
+// Sense returns measurements as °C and kPa.
+func (d *Dev) Sense(env *devices.Environment) error {
+	// start conversion for temperature
+	var cmd byte
+	cmd = 1 << ctrlStartConversion
+	cmd |= ctrlCommandTemperature << ctrlMeasurementControl
+
+	if err := d.d.Tx([]byte{regCtrlMeas, cmd}, nil); err != nil {
+		return err
+	}
+
+	time.Sleep(100 * time.Millisecond)
+
+	// read value
+	var tempBuf [2]byte
+	if err := d.readReg(regOutMSB, tempBuf[:]); err != nil {
+		return err
+	}
+
+	UT := uint16(tempBuf[0])<<8 | uint16(tempBuf[1])
+	temp := d.c.compensateTemp(UT)
+
+	// start conversion for pressure
+	cmd = 1 << ctrlStartConversion
+	cmd |= ctrlCommandPressure << ctrlMeasurementControl
+	cmd |= byte(d.os) << ctrlOversampling
+
+	if err := d.d.Tx([]byte{regCtrlMeas, cmd}, nil); err != nil {
+		return err
+	}
+
+	time.Sleep(100 * time.Millisecond)
+
+	// read value
+	var pressureBuf [3]byte
+	if err := d.readReg(regOutMSB, pressureBuf[:]); err != nil {
+		return err
+	}
+
+	UP := (int32(pressureBuf[0])<<16 + int32(pressureBuf[1])<<8 | int32(pressureBuf[2])) >> (8 - d.os)
+
+	pressure := d.c.compensatePressure(UP, int32(UT), d.os)
+
+	env.Temperature = devices.Celsius(temp * 100)
+	env.Pressure = devices.KPascal(pressure)
+	return nil
+}
+
+// Halt is a noop for the BMP180.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// Opts is optional options to pass to the constructor.
+type Opts struct {
+	Pressure Oversampling
+}
+
+const (
+	bmp180Address = 0x77 // the address of a BMP180 is fixed at 0x77
```

Comment:
You are in package bmp180, so prefixing a const with "bmp180" is redundant.

It was more of a nit than anything else, I don't mind much.